### PR TITLE
Alisha shreyas/admin dash

### PR DESCRIPTION
**Pull request by shreyas-thirumale**
# Pull Request

## Type of Change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update
- [ ] Refactoring (no functional changes)
- [ ] Performance improvement
- [ ] Other (please describe):

### Code Review Preparation
- [X] I have performed a self-review and reviewed the changed files
- [X] My code follows the project's style guidelines and passes linting
- [ ] I have added comments in hard-to-understand areas and updated documentation

### Testing (Ignore for the first PR)
- [X] I have added relevant tests and they pass locally
- [X] I have tested the changes manually, including edge cases

### Code Quality & Security
- [X] I have formatted my code and removed debugging/unused code
- [X] No TypeScript errors or linting issues
- [X] No sensitive information committed (API keys, passwords, etc.)
- [X] User inputs are validated and sanitized

### Dependencies & Deployment
- [ ] Dependencies are documented and package-lock.json is updated if needed
- [X] Changes work with Firebase emulator and environment variables are documentedRetry

## Screenshots (If it is a front end feature screenshot is required)

image

## Release Notes

* **New Features**
  * Redesigned admin home page with personalized welcome greeting and dashboard summaries (quick stats, warehouse history, pickups/deliveries, and upcoming shifts).
  * Added new schedule navigation link in the sidebar for admins.
  * Profile link now redirects to profile settings.
  * Updated sidebar layout with new visual indicator in the profile card.

### CODE REVIEW

File: components/icons/AdminWarehousePlus.tsx
Comment (constrictingsnake) [inline components/icons/AdminWarehousePlus.tsx:1]: instead of creating icons for plus and minus, you can just do it how warehouse history does it inline

File: components/icons/AdminWarehouseMinus.tsx
Comment (constrictingsnake) [inline components/icons/AdminWarehouseMinus.tsx:1]: instead of creating icons for plus and minus, you can just do it how warehouse history does it inline

File: components/homepage/AdminHomePage.tsx [hunk 1]
```diff
@@ -1,16 +1,37 @@
...
         <ProtectedRoute allow={["Admin"]}>
             <div className="h-full w-full flex flex-col font-family-roboto">
-                <div className="flex flex-1">
+                <div className="flex flex-1 min-h-0">
                     <SideNavbar />
-                    <div className="flex-1 bg-[#F7F7F7] pt-8 pb-4 px-6 flex flex-col">
-                        <span className="text-2xl text-primary font-extrabold block">
-                            Journeying to the Home Page! (Admin Home Page)
-                        </span>
+                    <div className="flex-1 flex flex-col overflow-auto pt-8 pb-4 px-6 gap-4 bg-linear-to-r from-[#F8FDFE] to-[#DAF2F5]">
```
Comment (constrictingsnake) [inline components/homepage/AdminHomePage.tsx]: there needs to be more padding on the bottom (the figma says 60 px so u can use pb-15)

File: components/homepage/AdminHomePage.tsx [hunk 2]
```diff
@@ -1,16 +1,37 @@
...
+                        <h1 className="text-4xl text-primary font-extrabold">
```
Comment (constrictingsnake) [inline components/homepage/AdminHomePage.tsx]: the figma requires the heading to have text-5xl

File: components/homepage/AdminHomePage.tsx [hunk 3]
```diff
@@ -1,16 +1,37 @@
...
+                            Welcome, {userData?.firstName ?? ""}!
+                        </h1>
+                        <div className="flex gap-4 flex-1 min-h-0">
+                            {/* Left column */}
+                            <div className="flex flex-col gap-4 w-[45%] min-w-0">
+                                <QuickStatsSummary />
```
Comment (constrictingsnake) [inline components/homepage/AdminHomePage.tsx]: wrap these components in divs that preserve their height (50-50 split according to the figma)

File: components/homepage/AdminHomePage.tsx [hunk 4]
```diff
@@ -1,16 +1,37 @@
...
+                                <WarehouseHistorySummary />
+                            </div>
+                            {/* Right column */}
+                            <div className="flex flex-col gap-4 flex-1 min-w-0">
+                                <PickupsDeliveriesSummary />
```
Comment (constrictingsnake) [inline components/homepage/AdminHomePage.tsx]: wrap these components in divs that preserve their height (50-50 split according to the figma)

File: components/homepage/AdminHomePage.tsx [hunk 5]
```diff
@@ -1,16 +1,37 @@
...
-                        <span className="text-2xl text-primary font-extrabold block">
-                            Journeying to the Home Page! (Admin Home Page)
-                        </span>
+                    <div className="flex-1 flex flex-col overflow-auto pt-8 pb-4 px-6 gap-4 bg-linear-to-r from-[#F8FDFE] to-[#DAF2F5]">
+                        <h1 className="text-4xl text-primary font-extrabold">
+                            Welcome, {userData?.firstName ?? ""}!
+                        </h1>
+                        <div className="flex gap-4 flex-1 min-h-0">
+                            {/* Left column */}
+                            <div className="flex flex-col gap-4 w-[45%] min-w-0">
```
Comment (constrictingsnake) [inline components/homepage/AdminHomePage.tsx]: the figma makes the right slightly larger than the left, so use the figma to get the correct split between the left and right side

File: components/homepage/AdminHomePage.tsx (hunk 1 above)
Comment (constrictingsnake) [reply components/homepage/AdminHomePage.tsx]: more padding on the left and right sides too (figma says around 40px so u can use the appropriate tailwind size)

Comment (constrictingsnake) [reply components/homepage/AdminHomePage.tsx]: and make the gap bigger to account for the space between the welcome text and the components

File: components/homepage/AdminCalendarSummary.tsx [hunk 6]
```diff
@@ -0,0 +1,86 @@
...
+
+    const upcomingEvents = useMemo(() => {
+        const now = new Date();
+        return [...allTB].filter((tb) => tb.startTime.toDate() >= now).sort((a,b)=> a.startTime.toDate().getTime() - b.startTime.toDate().getTime()).slice(0,4);
+
+    }, [allTB]);
+
+    return (
+        <div className="w-full rounded-2xl border border-[#E7E7E7] bg-white shadow-sm px-4 pb-4 flex flex-col gap-3">
+            <div className = "my-[1rem] font-bold text-lg"> Calendar </div>
```
Comment (constrictingsnake) [inline components/homepage/AdminCalendarSummary.tsx]: there needs to be a right aligned element that mentions how many of the shifts are unfilled (ex. "2/4 shifts unfilled"). also, in there should be more padding on the top

File: components/homepage/AdminCalendarSummary.tsx [hunk 7]
```diff
@@ -0,0 +1,86 @@
...
+        })
+        .replace(":00", "")
+        .toLowerCase()
+        .replaceAll(" ", "");
+}
+
+export default function AdminCalendarSummary() {
+    const { allTB, isLoading } = useTimeBlocks();
+
+    const upcomingEvents = useMemo(() => {
```
Comment (constrictingsnake) [inline components/homepage/AdminCalendarSummary.tsx:51]: the shifts should be grouped by shifts that occur on the same day, and they should be shown in a singular day slot

File: components/homepage/AdminCalendarSummary.tsx [hunk 8]
```diff
@@ -0,0 +1,86 @@
...
+    return (
+        <div className="w-full rounded-2xl border border-[#E7E7E7] bg-white shadow-sm px-4 pb-4 flex flex-col gap-3">
+            <div className = "my-[1rem] font-bold text-lg"> Calendar </div>
+            <div className="border-b border-[#E3E3E3]" />
+            {isLoading ? (
+                <p className = "flex justify-center"> Loading... </p>
+            ) : upcomingEvents.length === 0 ? (
+                <p className = "flex justify-center">No Upcoming Timeblocks</p>
+            ) : (
+                <div className = " flex flex-col">
```
Comment (constrictingsnake) [inline components/homepage/AdminCalendarSummary.tsx]: within the entire component, the styling should match the figma more. comparing the current implementation to the figma, the days themselves are rendered differently, with the figma having each day on its own. the dates are also shown differently. the colored circles representing the type of shift are smaller. the icons on the right showing the volunteers and drivers should be right aligned. as a starting point, look at how the PickupsDeliveriesSummary component does it.

File: components/homepage/AdminCalendarSummary.tsx [hunk 9]
```diff
@@ -0,0 +1,86 @@
...
+                                <span> {formatTime(tb.startTime)}-{formatTime(tb.endTime)}</span>
+                                <span> {tb.type}</span>
+                                <span className={`flex items-center gap-[0.25rem] ${lowVol ? "text-[#E16060]" : "text-[#000000]"}`}>
+                                    <AdminCalendarPeople fill={lowVol ? "#E16060" : "#000000"}/>
+                                    {volCount}
+                                </span>
+                                {lowDrive ? (
+                                <span className={`flex items-center gap-[0.25rem] ${lowDrive ? "text-[#E16060]" : "text-[#000000]"}`}>
+                                    <AdminCalendarDriver fill={lowDrive ? "#E16060" : "#000000"}/>
+                                    {volCount}
```
Comment (constrictingsnake) [inline components/homepage/AdminCalendarSummary.tsx]: the driver icon currently shows the volunter count instead of the driver count

File: components/homepage/AdminCalendarSummary.tsx [hunk 10]
```diff
@@ -0,0 +1,86 @@
...
+    const { allTB, isLoading } = useTimeBlocks();
+
+    const upcomingEvents = useMemo(() => {
+        const now = new Date();
+        return [...allTB].filter((tb) => tb.startTime.toDate() >= now).sort((a,b)=> a.startTime.toDate().getTime() - b.startTime.toDate().getTime()).slice(0,4);
+
+    }, [allTB]);
+
+    return (
+        <div className="w-full rounded-2xl border border-[#E7E7E7] bg-white shadow-sm px-4 pb-4 flex flex-col gap-3">
```
Comment (constrictingsnake) [inline components/homepage/AdminCalendarSummary.tsx]: make sure the typography of the text within the component matches whats stated in the figma

File: components/homepage/PickupsDeliveriesSummary.tsx [hunk 11]
```diff
@@ -0,0 +1,95 @@
...
+    const d = new Date(tb.startTime.seconds * 1000);
+    const now = new Date();
+    return (
+        d.getFullYear() === now.getFullYear() &&
+        d.getMonth() === now.getMonth() &&
+        d.getDate() === now.getDate()
+    );
+}
+
+function formatTimeRange(tb: TimeBlock): string {
```
Comment (constrictingsnake) [inline components/homepage/PickupsDeliveriesSummary.tsx]: the times don't need to be this detailed, look at how its done in admin calendar summary

File: components/homepage/PickupsDeliveriesSummary.tsx [hunk 12]
```diff
@@ -0,0 +1,95 @@
...
+                const dr = task as DonationRequest;
+                return { key: `${tb.id}-${dr.id}`, timeRange, name: `${dr.donor.firstName} ${dr.donor.lastName}`, location: dr.donor.address?.city ?? "", type: "P" as const };
+            }
+        });
+    });
+
+    return (
+        <div className="h-[26.5%] bg-white rounded-2xl border border-light-border p-5 flex flex-col gap-3">
+            <div className="flex items-center justify-between">
+                <span className="text-lg font-bold text-text-1">Upcoming pickups/deliveries</span>
```
Comment (constrictingsnake) [inline components/homepage/PickupsDeliveriesSummary.tsx]: Heading font size doesn't match the figma

File: components/homepage/PickupsDeliveriesSummary.tsx [hunk 13]
```diff
@@ -0,0 +1,95 @@
...
+                <span className="text-sm text-gray-400">Loading...</span>
+            ) : isError ? (
+                <span className="text-sm text-gray-400">Unable to load pickups or deliveries</span>
+            ) : rows.length === 0 ? (
+                <span className="text-sm text-gray-400">No pickups or deliveries today</span>
+            ) : (
+                <div className="flex flex-col gap-2">
+                    {rows.map((row) => (
+                        <div key={row.key} className="flex items-center gap-3 border border-light-border px-3 py-2">
+                            <span className={`w-8 h-8 rounded-full flex items-center justify-center text-sm font-semibold shrink-0 ${row.type === "P" ? "bg-[#D6E8F0] text-[#4A8FA8]" : "bg-[#F5E0E0] text-[#A87070]"}`}>{row.type}</span>
```
Comment (constrictingsnake) [inline components/homepage/PickupsDeliveriesSummary.tsx]: the circles on the figma are 28px, which would be w-7 h-7

File: components/homepage/PickupsDeliveriesSummary.tsx [hunk 14]
```diff
@@ -0,0 +1,95 @@
...
+                            <span className="w-20 shrink-0 text-xs text-gray-500">{row.timeRange}</span>
```
Comment (constrictingsnake) [inline components/homepage/PickupsDeliveriesSummary.tsx]: the row text is too small

File: components/homepage/PickupsDeliveriesSummary.tsx [hunk 15]
```diff
@@ -0,0 +1,95 @@
...
+                            <span className="flex items-center gap-1 flex-1 text-xs text-gray-500"><PersonIcon />{row.name}</span>
```
Comment (constrictingsnake) [inline components/homepage/PickupsDeliveriesSummary.tsx]: flex-1 incorrectly pushes the location to the far right, when the location isn't right aligned

File: components/homepage/PickupsDeliveriesSummary.tsx [hunk 16]
```diff
@@ -0,0 +1,95 @@
...
+            {isLoading ? (
+                <span className="text-sm text-gray-400">Loading...</span>
+            ) : isError ? (
+                <span className="text-sm text-gray-400">Unable to load pickups or deliveries</span>
+            ) : rows.length === 0 ? (
+                <span className="text-sm text-gray-400">No pickups or deliveries today</span>
+            ) : (
+                <div className="flex flex-col gap-2">
+                    {rows.map((row) => (
+                        <div key={row.key} className="flex items-center gap-3 border border-light-border px-3 py-2">
```
Comment (constrictingsnake) [inline components/homepage/PickupsDeliveriesSummary.tsx]: the corners need to be rounded

File: components/homepage/PickupsDeliveriesSummary.tsx [hunk 17]
```diff
@@ -0,0 +1,95 @@
...
+                return { key: `${tb.id}-${cr.id}`, timeRange, name: `${cr.client.firstName} ${cr.client.lastName}`, location: cr.client.address?.city ?? "", type: "D" as const };
+            } else {
+                const dr = task as DonationRequest;
+                return { key: `${tb.id}-${dr.id}`, timeRange, name: `${dr.donor.firstName} ${dr.donor.lastName}`, location: dr.donor.address?.city ?? "", type: "P" as const };
+            }
+        });
+    });
+
+    return (
+        <div className="h-[26.5%] bg-white rounded-2xl border border-light-border p-5 flex flex-col gap-3">
```
Comment (constrictingsnake) [inline components/homepage/PickupsDeliveriesSummary.tsx]: the background isn't white, it should match the figma

File: components/homepage/PickupsDeliveriesSummary.tsx [hunk 18]
```diff
@@ -0,0 +1,95 @@
...
+            </div>
+            {isLoading ? (
+                <span className="text-sm text-gray-400">Loading...</span>
+            ) : isError ? (
+                <span className="text-sm text-gray-400">Unable to load pickups or deliveries</span>
+            ) : rows.length === 0 ? (
+                <span className="text-sm text-gray-400">No pickups or deliveries today</span>
+            ) : (
+                <div className="flex flex-col gap-2">
+                    {rows.map((row) => (
```
Comment (constrictingsnake) [inline components/homepage/PickupsDeliveriesSummary.tsx]: only 4 pickups/deliveries need to be shown in the component, but the count on the top should still accurately reflect the number of pickups/deliveries scheduled during that day

File: components/homepage/QuickStatsSummary.tsx [hunk 19]
```diff
@@ -0,0 +1,89 @@
...
+    const itemDate = new Date(seconds * 1000);
+    const now = new Date();
+    return (
+        itemDate.getFullYear() === now.getFullYear() &&
+        itemDate.getMonth() === now.getMonth() &&
+        itemDate.getDate() === now.getDate()
+    );
+}
+
+function ArrowIcon() {
```
Comment (constrictingsnake) [inline components/homepage/QuickStatsSummary.tsx]: the arrow icon is too small

File: components/homepage/QuickStatsSummary.tsx [hunk 20]
```diff
@@ -0,0 +1,89 @@
...
+    label: string;
+    count: number;
+    newCount?: number;
+    icon: React.ReactNode;
+    iconClassName?: string;
+};
+
+function StatCard({ label, count, newCount, icon, iconClassName }: StatCardProps) {
+    return (
+        <div className="bg-white rounded-2xl border border-light-border p-4 flex flex-col gap-1 relative overflow-hidden h-40">
```
Comment (constrictingsnake) [inline components/homepage/QuickStatsSummary.tsx]: the border is too thin; the gap-1 causes the spacing to be too little between the title row and the number

File: components/homepage/QuickStatsSummary.tsx [hunk 21]
```diff
@@ -0,0 +1,89 @@
...
+            <div className="flex items-center justify-between">
+                <span className="text-sm font-semibold text-text-1">{label}</span>
```
Comment (constrictingsnake) [inline components/homepage/QuickStatsSummary.tsx]: the typography of the label doesn't match the figma

File: components/homepage/QuickStatsSummary.tsx [hunk 22]
```diff
@@ -0,0 +1,89 @@
...
+                <span className="text-text-1 opacity-70"><ArrowIcon /></span>
+            </div>
+            <span className="text-5xl font-normal text-text-1 mt-1">{count}</span>
+            <span className="text-xs text-gray-400 mt-auto">
```
Comment (constrictingsnake) [inline components/homepage/QuickStatsSummary.tsx]: text xs is too small, refer to the figma

File: components/homepage/QuickStatsSummary.tsx [hunk 23]
```diff
@@ -0,0 +1,89 @@
...
+                {newCount !== undefined ? `${newCount} new today` : ""}
+            </span>
+            <div className={`absolute opacity-[0.15] text-[#a0c4d8] text-[5rem] ${iconClassName ?? "bottom-[-1.25rem] right-[-1.25rem]"}`}>
```
Comment (constrictingsnake) [inline components/homepage/QuickStatsSummary.tsx]: the background icon is too faint

File: components/homepage/QuickStatsSummary.tsx [hunk 24]
```diff
@@ -0,0 +1,89 @@
...
+    icon: React.ReactNode;
+    iconClassName?: string;
+};
+
+function StatCard({ label, count, newCount, icon, iconClassName }: StatCardProps) {
+    return (
+        <div className="bg-white rounded-2xl border border-light-border p-4 flex flex-col gap-1 relative overflow-hidden h-40">
+            <div className="flex items-center justify-between">
+                <span className="text-sm font-semibold text-text-1">{label}</span>
+                <span className="text-text-1 opacity-70"><ArrowIcon /></span>
```
Comment (constrictingsnake) [inline components/homepage/QuickStatsSummary.tsx]: the opacity-70 is unnecessary here

File: components/homepage/QuickStatsSummary.tsx [hunk 25]
```diff
@@ -0,0 +1,89 @@
...
+
+    if (isLoading) {
+        return <div className="grid grid-cols-2 gap-3 animate-pulse">
+            {[...Array(4)].map((_, i) => <div key={i} className="bg-white rounded-sm border border-light-border h-28" />)}
+        </div>;
+    }
+
+    return (
+        <div className="grid grid-cols-2 gap-3">
+            <StatCard label="Low stock items" count={lowStockCount} newCount={0} icon={<InventoryIcon />} iconClassName="bottom-[-1.25rem] right-[-1.25rem]" />
```
Comment (constrictingsnake) [inline components/homepage/QuickStatsSummary.tsx]: use warehouse history logs to compute if there's any new categories with low counts within the past day rather than hard coding it to be 0

File: components/homepage/QuickStatsSummary.tsx [hunk 26]
```diff
@@ -0,0 +1,89 @@
...
+            <StatCard label="Donation requests" count={notReviewedDR.length} newCount={newDR} icon={<DonorRequestsIcon />} iconClassName="bottom-[-0.5rem] right-[-1rem]" />
+            <StatCard label="Account requests" count={accountRequests.length} newCount={0} icon={<UserManagementIcon />} iconClassName="bottom-[-0.25rem] right-[-1rem]" />
```
Comment (constrictingsnake) [inline components/homepage/QuickStatsSummary.tsx]: hard-coding the account requests is something that needs to be addressed, we will reach out with details

File: components/homepage/QuickStatsSummary.tsx [hunk 27]
```diff
@@ -0,0 +1,89 @@
...
+    const notReviewedCR = clientRequests.filter((cr) => cr.status === "Not Reviewed");
+    const newCR = notReviewedCR.filter((cr) => isFromToday(cr.date.seconds)).length;
+
+    const notReviewedDR = donationRequests.filter((dr) =>
+        dr.items.some((item) => item.status === "Not Reviewed")
+    );
+    const newDR = notReviewedDR.filter((dr) => isFromToday(dr.date.seconds)).length;
+
+    if (isLoading) {
+        return <div className="grid grid-cols-2 gap-3 animate-pulse">
```
Comment (constrictingsnake) [inline components/homepage/QuickStatsSummary.tsx]: the loading skeletons should match the size of the actual cards

File: components/homepage/QuickStatsSummary.tsx [hunk 28]
```diff
@@ -0,0 +1,89 @@
...
+                {newCount !== undefined ? `${newCount} new today` : ""}
+            </span>
+            <div className={`absolute opacity-[0.15] text-[#a0c4d8] text-[5rem] ${iconClassName ?? "bottom-[-1.25rem] right-[-1.25rem]"}`}>
+                {icon}
+            </div>
+        </div>
+    );
+}
+
+export function QuickStatsSummary() {
```
Comment (constrictingsnake) [inline components/homepage/QuickStatsSummary.tsx:53]: add error handling

File: components/homepage/PickupsDeliveriesSummary.tsx (hunk 17 above)
Comment (constrictingsnake) [reply components/homepage/PickupsDeliveriesSummary.tsx]: and the border needs to be a bit thicker

File: components/homepage/AdminCalendarSummary.tsx (hunk 10 above)
Comment (constrictingsnake) [reply components/homepage/AdminCalendarSummary.tsx]: and the border needs to be thicker

File: components/homepage/WarehouseHistorySummary.tsx [hunk 29]
```diff
@@ -0,0 +1,65 @@
...
+        [warehouseChanges]
+    );
+    const twoDaysAfter = useMemo(() =>
+        sortedChanges.filter(
+            (c) => Date.now() - c.timestamp.toDate().getTime() <= 48 * 60 * 60 * 1000,
+        ).length,
+        [sortedChanges]
+    );
+
+    const mostRecent = sortedChanges.slice(0,5);
```
Comment (constrictingsnake) [inline components/homepage/WarehouseHistorySummary.tsx]: you can use a useMemo here as well

File: components/homepage/WarehouseHistorySummary.tsx [hunk 30]
```diff
@@ -0,0 +1,65 @@
...
+                <p className="text-sm text-gray-400">No inventory changes were made</p>
+            ) : (
+                <div className="flex flex-col gap-2">
+                    {mostRecent.map((c) => {
+                        const diff = c.change.newQuantity - c.change.oldQuantity;
+                        const isPositive = diff > 0;
+                        return (
+                            <div key={c.id} className="flex items-center gap-3 border border-light-border px-3 py-2">
+                                <span className="shrink-0">
+                                    {isPositive ? <AdminWarehousePlus /> : <AdminWarehouseMinus />}
```
Comment (constrictingsnake) [inline components/homepage/WarehouseHistorySummary.tsx]: the control panel uses inline svgs, which you should use instead of the icon for sake of consistency

File: components/homepage/WarehouseHistorySummary.tsx [hunk 31]
```diff
@@ -0,0 +1,65 @@
...
+                <p className="text-sm text-gray-400">Loading...</p>
+            ): mostRecent.length === 0 ? (
+                <p className="text-sm text-gray-400">No inventory changes were made</p>
+            ) : (
+                <div className="flex flex-col gap-2">
+                    {mostRecent.map((c) => {
+                        const diff = c.change.newQuantity - c.change.oldQuantity;
+                        const isPositive = diff > 0;
+                        return (
+                            <div key={c.id} className="flex items-center gap-3 border border-light-border px-3 py-2">
```
Comment (constrictingsnake) [inline components/homepage/WarehouseHistorySummary.tsx]: the rows should be rounded

File: components/homepage/WarehouseHistorySummary.tsx [hunk 32]
```diff
@@ -0,0 +1,65 @@
...
+                                <span className="shrink-0">
+                                    {isPositive ? <AdminWarehousePlus /> : <AdminWarehouseMinus />}
+                                </span>
+                                <span className="text-xs text-gray-400 shrink-0">
+                                    {c.timestamp.toDate().toLocaleDateString("en-US", { weekday: "short", month: "short", day: "numeric", hour: "numeric", minute: "2-digit" })}
```
Comment (constrictingsnake) [inline components/homepage/WarehouseHistorySummary.tsx]: Match the time with the time that's on the figma; no commas and the am/pm is lowercase

File: components/homepage/WarehouseHistorySummary.tsx [hunk 33]
```diff
@@ -0,0 +1,65 @@
...
+        sortedChanges.filter(
+            (c) => Date.now() - c.timestamp.toDate().getTime() <= 48 * 60 * 60 * 1000,
+        ).length,
+        [sortedChanges]
+    );
+
+    const mostRecent = sortedChanges.slice(0,5);
+
+    return (
+        <div className="w-full rounded-2xl border border-[#E7E7E7] bg-white shadow-sm p-4 flex flex-col gap-3">
```
Comment (constrictingsnake) [inline components/homepage/WarehouseHistorySummary.tsx]: the background isn't supposed to be white, you can get the color from the figma

File: components/homepage/WarehouseHistorySummary.tsx [hunk 34]
```diff
@@ -0,0 +1,65 @@
...
+            <div className="flex flex-row place-content-between items-center">
+                <span className="font-bold text-lg text-text-1">
```
Comment (constrictingsnake) [inline components/homepage/WarehouseHistorySummary.tsx]: the heading typography is inaccurate; refer to the figma

File: components/homepage/QuickStatsSummary.tsx (hunk 19 above)
Comment (constrictingsnake) [inline components/homepage/QuickStatsSummary.tsx]: the arrow icon is too small

File: components/homepage/QuickStatsSummary.tsx (hunk 20 above)
Comment (constrictingsnake) [inline components/homepage/QuickStatsSummary.tsx]: the border is too thin; the gap-1 causes the spacing to be too little between the title row and the number

File: components/homepage/QuickStatsSummary.tsx (hunk 21 above)
Comment (constrictingsnake) [inline components/homepage/QuickStatsSummary.tsx]: the typography of the label doesn't match the figma

File: components/homepage/QuickStatsSummary.tsx (hunk 24 above)
Comment (constrictingsnake) [inline components/homepage/QuickStatsSummary.tsx]: the opacity-70 is unnecessary here

File: components/homepage/QuickStatsSummary.tsx (hunk 22 above)
Comment (constrictingsnake) [inline components/homepage/QuickStatsSummary.tsx]: text xs is too small, refer to the figma

File: components/homepage/QuickStatsSummary.tsx (hunk 23 above)
Comment (constrictingsnake) [inline components/homepage/QuickStatsSummary.tsx]: the background icon is too faint

File: components/homepage/QuickStatsSummary.tsx (hunk 28 above)
Comment (constrictingsnake) [inline components/homepage/QuickStatsSummary.tsx:53]: add error handling to make sure that the user know if something fails

File: components/homepage/QuickStatsSummary.tsx (hunk 27 above)
Comment (constrictingsnake) [inline components/homepage/QuickStatsSummary.tsx]: the loading skeletons should match the size of the actual cards

File: components/homepage/QuickStatsSummary.tsx (hunk 25 above)
Comment (constrictingsnake) [inline components/homepage/QuickStatsSummary.tsx]: use warehouse history logs to compute if there's any new categories with low counts within the past day rather than hard coding it to be 0

File: components/homepage/QuickStatsSummary.tsx (hunk 26 above)
Comment (constrictingsnake) [inline components/homepage/QuickStatsSummary.tsx]: account requests don't need the text at the bottom, so you don't have to worry about this

File: components/homepage/WarehouseHistorySummary.tsx (hunk 33 above)
Comment (constrictingsnake) [inline components/homepage/WarehouseHistorySummary.tsx]: the background isn't supposed to be white, you can get the color from the figma

File: components/homepage/WarehouseHistorySummary.tsx (hunk 34 above)
Comment (constrictingsnake) [inline components/homepage/WarehouseHistorySummary.tsx]: the heading typography is inaccurate; refer to the figma

File: components/homepage/WarehouseHistorySummary.tsx (hunk 31 above)
Comment (constrictingsnake) [inline components/homepage/WarehouseHistorySummary.tsx]: the rows should be rounded

File: components/homepage/WarehouseHistorySummary.tsx (hunk 30 above)
Comment (constrictingsnake) [inline components/homepage/WarehouseHistorySummary.tsx]: the control panel uses inline svgs, which you should use instead of the icon for sake of consistency

File: components/homepage/WarehouseHistorySummary.tsx (hunk 32 above)
Comment (constrictingsnake) [inline components/homepage/WarehouseHistorySummary.tsx]: Match the time with the time that's on the figma; no commas and the am/pm is lowercase

File: components/icons/AdminCalendarDriver.tsx
Comment (constrictingsnake) [inline components/icons/AdminCalendarDriver.tsx:1]: its calendar not calandar

File: components/homepage/PickupsDeliveriesSummary.tsx (hunk 17 above)
Comment (constrictingsnake) [reply components/homepage/PickupsDeliveriesSummary.tsx]: also h-[26.5%] isn't the best approach with percentages, you can get away with h-full so that it fills whatever wrapper the homepage puts around it

File: components/icons/DogPeekingIcon.tsx
Comment (constrictingsnake) [inline components/icons/DogPeekingIcon.tsx:1]: instead of using dashes here for properties like "stroke-width" etc, use camelcase to avoid errors (like "strokeWidth" etc

**Review by constrictingsnake — CHANGES_REQUESTED** [review]

**Review by joelchem — APPROVED** [review]
